flang/lib/Lower/Bridge.cpp

```diff
@@ -3807,16 +3807,30 @@ class FirConverter : public Fortran::lower::AbstractConverter {
     return temps;
   }
 
+  static bool isDeviceContext(fir::FirOpBuilder &builder) {
+    if (builder.getRegion().getParentOfType<fir::CUDAKernelOp>())
+      return true;
+    if (auto funcOp =
+            builder.getRegion().getParentOfType<mlir::func::FuncOp>()) {
+      if (auto cudaProcAttr =
+              funcOp.getOperation()->getAttrOfType<fir::CUDAProcAttributeAttr>(
+                  fir::getCUDAAttrName())) {
+        return cudaProcAttr.getValue() != fir::CUDAProcAttribute::Host;
```

Comment:
Can you please clarify what will happen for a `host,device` function?  It seems that we will treat it as "device context", and will not generate the data transfer operation.

I suppose there will have to be a pass that creates distinct host and device copies of the `host,device` function, and so we should be able to handle `fir.cuda_data_transfer` during the cloning. So should we still generate `fir.cuda_data_transfer` for "maybe host context" cases?

Reply:
Yes, there would be a pass that create the device func part. I guess the easiest is to still generate the data transfer for those and as you mentioned, we will handle them during the host/device func creation. 

I'll update the code to represent that and add some comment.